### PR TITLE
React Calculator Logic

### PR DESCRIPTION
**Pull request by kwizl**
# Project Logic

- In this milestone, two JavaScript modules are implemented: calculate.js and operate.js These two modules will be responsible for all the calculations. There are npm packages such as `Big.js` that are used to help in calculating big numbers.

### CODE REVIEW

**Comment by Torres-ssf** [comment]
Hello @kwizl :hand: 
It seems that you selected the wrong project name when sending the code review request. The request is related to the Milestone 4 from the Bookstore App. Please kindly submit a code review request to Calcular Project milestone 4 as I will invalidate this request related to the Bookstore App

File: src/logic/operate.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+const operate = (numberOne, numberTwo, operation) => {
+  if (operation) {
+    if (operation === '+') {
+      return Big(numberOne + numberTwo);
+    }
+
+    if (operation === '') {
+      return Big(numberOne - numberTwo);
+    }
```
Comment (Torres-ssf) [inline src/logic/operate.js]: - [x] It seems the `-` symbol is missing at the if statement

File: src/logic/operate.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+    if (operation === '-') {
+      return Big(numberOne * numberTwo);
+    }
```
Comment (Torres-ssf) [inline src/logic/operate.js]: - [x] Here we need to the `x` instead of `-`

File: src/logic/operate.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+    if (operation === '÷') {
+      return Big(numberOne / numberTwo);
+    }
+
+    if (operation === 'x') {
+      return Big((numberTwo * 100) / numberOne);
+    }
```
Comment (Torres-ssf) [inline src/logic/operate.js]: - [x] We need to use the symbol to the modifier operator here: `%`

- [x] Te modifier operator should turns the number into a percentage base `e.g. 95 % == 0.95`. We can achieve that by simply 
```JS
if (operation === 'x') {
      return Big((numberOne / 100);
    }
```

File: src/logic/calculate.js [hunk 4]
```diff
@@ -0,0 +1,49 @@
+import Big from 'big.js';
+import operate from './operate';
+
+const calculate = (dataObject, buttonName) => {
+  const { total, next, operation } = dataObject;
+  const operations = ['+', '-', 'x', '÷', '%'];
+
+  if (buttonName === 'A/C') {
+    return {
```
Comment (Torres-ssf) [inline src/logic/calculate.js:9]: - [x] We also need to verify cases where the user presses the dot button `.`

**Review by Torres-ssf — CHANGES_REQUESTED** [review]
### Milestone approved :+1: 
Hi @kwizl :hand: 
Great job on this milestone so far :clap: 
But we need to make some changes before moving on :smile: 

- [x] Please check my comments below.

Feel free to message me on Slack if you have any questions.
Happy coding :computer: 
**Sergio Torres**

**Review by whiz25 — APPROVED** [review]
## Approved ✅ 
Nice work @kwizl . You fixed operations in the `operate.js` file as suggested by the previous reviewer. This PR looks good to me. Merge and proceed to the next milestone.


### Martin Afani

**Pull request merged**